docs/markdown/Reference-tables.md

```diff
@@ -42,6 +43,7 @@ set in the cross file.
 | x86                 | 32 bit x86 processor  |
 | x86_64              | 64 bit x86 processor  |
 | arm                 | 32 bit ARM processor |
+| e2k                 | MCST Elbrus processor |
```

Comment:
Yes, it's the exactly same one that is available since automake 1.15.1.
About Python, I'm not sure (for example, in python 3.6.5 sources, I can't find any `e2k` substring in text files, except for automake-related `config.guess` and `config.sub`).

These processors are 64-bit from the beginning; but they support execution of 32-bit programs as well (provided that needed 32-bit libraries are supplied), for example, it is subject for non-portable programs that, say, treat unsigned int as uint32_t. It's nearly identical situation as with x86_64: it supports execution of both 64-bit and 32-bit code, even if OS kernel and CPU architecture is purely 64-bit (but with the exception that for x86_64, there is hardware 32-bit counterpart exist (x86); but there is no such counterpart for e2k).

Reply:
The detection is done with `platform.machine().lower()` in `detect_cpu_family`. Can you test that? If it returns the correct value, then everything should be fine. If not, please add the necessary code to return the correct value.

Reply:
Ping. We are approaching dev freeze for the next release and it would be great to get this in before that.

Reply:
I'll see this today or tomorrow (my elbrus testbench was busy in stability testing until yesterday's evening).

Reply:
As far as I see, it's all ok with this now:
```
The Meson build system
Version: 0.45.0.dev1
Source dir: /home/molchan_i/src/meson
Build dir: /home/molchan_i/src/meson/0
Build type: native build
Project name: simple
Native C compiler: cc (lcc 1.23.6 "lcc:1.23.06:Mar--4-2018:e2k-v4-linux")
Build machine cpu family: e2k
Build machine cpu: e8c
```
Note `e2k` in family and `e8c` in cpu (`e2k` was in both fields before commit 495f47b); I had to make more precise cpu detection due to the fact that e2k family has some different CPUs, like `e3m`, `e3s`, `e2c+`, `e2cv`, `e2c1`, `e2c1+`, `e8c`, `e12c`, and `e16c`.